consul/src/main/scala/io/buoyant/consul/CatalogNamer.scala

```diff
@@ -83,55 +93,58 @@ class CatalogNamer(
   }
 
   /**
-   * Contains all cached serviceNodes responses for a particular serviceName in a particular datacenter
+   * Contains all cached serviceNodes responses for a particular serviceName
+   * in a particular datacenter
    */
   case class SvcCache(datacenter: String, key: SvcKey, updateableAddr: VarUp[Addr]) {
 
     def addrs: VarUp[Addr] = updateableAddr
 
-    var index = "0"
     val _ = init()
 
-    def setIndex(idx: String) = {
-      index = idx
-    }
-
-    def mkRequest(): Future[Seq[ServiceNode]] =
-      api
-        .serviceNodes(key.name, datacenter = Some(datacenter), tag = key.tag, blockingIndex = Some(index), retry = true)
-        .map { indexedNodes =>
-          indexedNodes.index.foreach(setIndex)
-          indexedNodes.value
-        }
+    def mkRequest(index: Option[String]) =
```

Comment:
Yeah, a bit of it. I just saw a way to get rid of variable and changed the code a bit. But can put this into a separate PR.